alphabase/peptide/smiles.py

```diff
@@ -0,0 +1,21 @@
+import functools
+
+def requires_rdkit(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            from rdkit import Chem
```

Comment:
is there another option to find out that the package is there, without actually importing it? this would avoid the side effect of this decorator

Reply:
also, this will only work if the imports are done within the functions not for top-level imports, right?

Reply:
You could use `importlib.util.find_spec("rdkit") is not None` which still invokes the import machinery to some degree.

On the second point, I think there is no way around this.
If you have a module level import you will always fail during initialization. 
Maybe it's possible with some heavy import magic but I'm not aware.

Reply:
okay, but can we then not just catch the import error in the `smiles` module, and change it to the user-friendly error message?

Reply:
Yes, I prefer to catch ImportError once this module is imported